stellar-contract-env-host/src/host.rs

```diff
@@ -1008,10 +1008,34 @@ impl CheckedEnv for Host {
     }
 
     fn compute_hash_sha256(&self, x: Object) -> Result<Object, HostError> {
-        todo!()
+        use sha2::{Digest, Sha256};
+        let hash = self.visit_obj(x, |bin: &Vec<u8>| {
+            Ok(Sha256::digest(bin).as_slice().to_vec())
+        })?;
+
+        if hash.len() != 32 {
+            return Err(HostError::General("incorrect hash size"));
+        }
+
+        Ok(self.add_host_object(hash)?.into())
```

Comment:
It looks like this returns a SCO_BINARY, but the fn should return an SCO_HASH. I don't think the hash type is added yet. cc @jayz22 who I think is adding hash in https://github.com/stellar/rs-stellar-xdr/pull/87.

Reply:
Yes that's in the process of being added. Once merged, I can update it.

Reply:
I still don't think the hash/public key/binary thing is settled. Let's stick with binary as much as possible until we see a reason to do the other things (because doing the other things requires adding additional functions, so sticking with binary is the path of least resistance for now).

Reply:
If we're debating adding Hash/PublicKey, let's resolve that, because we're adding XDR types in https://github.com/stellar/rs-stellar-xdr/pull/87 which will lead to work to add contract spec support for them, as well as sdk types. I'd like to resolve this so we don't build all these things out if we don't need them.

Reply:
Agree. 
This issue was previously discussed in the dev-group, in particular @graydon's comment here:
https://groups.google.com/g/stellar-dev/c/qpq4VS1ft3k/m/TUqAzXZKBQAJ explains the rationale for including these types.
I can repost in the Discord chat for more discussion.

Reply:
Found it: https://discord.com/channels/897514728459468821/979885313377845278 (jump-cannon-dev thread "Should CONTRACT CODE remain different").